File: tests/fixtures/sample.py
# Module-level comment to be stripped.



def add(a, b):
    # inline note about x
    return a + b  # end-of-line note


class Counter:
    """Docstring with a # inside that must survive."""

    def __init__(self):
        self.n = 0

    def increment(self):
        self.n += 1
        return self.n


message = "this # is not a comment"
multi = """triple-quoted
spanning lines
with # inside still preserved"""

c = Counter()
add(c.increment(), 41)

File: tests/fixtures/sample.c
/* top-level block comment */
#include <stdio.h>
#include <stdlib.h>

// line comment about MAX
#define MAX 100
#define GREET(name) printf("Hello, %s\n", name)

/* A struct with a closing brace } inside a string below */
typedef struct {
    int id;
    char label[32];
} Item;

/*
 * multi-line block comment
 * explaining add()
 */
int add(int a, int b) {
    // inline note
    return a + b;
}

int main(void) {
    int result = add(3, 4);
    /* another comment */
    char *fmt = "result = %d }"; /* } in string must not end a block */
    printf(fmt, result);
    return 0;
}
